mobius-extras/src/main/java/com/spotify/mobius/extras/connections/ContramapConnection.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius.extras.connections;
+
+import static com.spotify.mobius.internal_util.Preconditions.checkNotNull;
+
+import com.spotify.mobius.Connectable;
+import com.spotify.mobius.Connection;
+import com.spotify.mobius.functions.Consumer;
+import com.spotify.mobius.functions.Function;
+
+/** A {@link Connection} implementation that does contravariant mapping. */
+public class ContramapConnection<A, B, C> implements Connection<B> {
```

Comment:
Should the `Connection` implementations not be package-private? Is there a reason to make these implementations a part of the public API?

Reply:
@pettermahlen What we really need is module-private :) I can make the change if you want, if you'll have enough time to approve :D